### PR TITLE
Add CLA Signature bot

### PR DESCRIPTION
**Pull request by Gudahtt**
The CLA signature bot will check the authors of each PR to ensure they have all signed the CLA. If any authors still need to sign the CLA, it will leave a comment explaining how it can be signed, and will check back upon each comment to see if it has been signed.

The bot used is `Metamask/cla-signature-bot`, which is a fork of `Roblox/cla-signature-bot`. The fork has a couple of improvements, and it updated the PR comment text to be more appropriate for our usage.

Currently the only whitelisted user is `dependabot`, meaning that even ConsenSys employees will need to sign the CLA. We could add all employees to the whitelist, but I figured it'd be easier to get everyone to sign individually rather than maintaining this list here.

The signatures are stored in `cla.json` on the `cla-signatures` branch, which is in this repository as a distinct root. We can consider moving this to a separate repository in the future - this was just easier to setup.

### CODE REVIEW

**Comment by Gudahtt** [comment]
Note that actions haven't yet been enabled on this repo, and even if they had, it wouldn't work properly yet because of my use of the `pull_request_target` event trigger (which only uses actions that are in the base branch).

**Comment by Gudahtt** [comment]
Currently the workflow for first-time contributors/signers requires the action to be manually re-run unfortunately. It's supposed to update automatically after the author signs, but it doesn't seem to be working correctly. Instead of re-running, the job gets queued up again but never executes (probably a GitHub bug? :thinking:). Manually re-running once per contributor isn't terrible though.

**Comment by Gudahtt** [comment]
I have just added a branch protection rule for the `cla-signatures` branch. I couldn't enable any of the options like requiring status checks (as the action must be allowed to push directly to that branch), but at least it prevents deletion and force pushes and non-linear history.

**Comment by Gudahtt** [comment]
Note that actions triggered by the `pull_request_target` event are automatically added as status checks. We can make this a required status check after this is merged.

<details>
<summary>Screenshot:</summary>

status_check

</details>

File: .github/workflows/cla.yml
Comment (rekmarks) [inline .github/workflows/cla.yml:21]: Is this hardcoded? We should move to `safelist`/`allowlist` if possible.

**Review by rekmarks — DISMISSED** [review]
LGTM, just one language-related nit.

File: .github/workflows/cla.yml
Comment (Gudahtt) [reply .github/workflows/cla.yml:21]: It's a custom property used by the action. We can certainly update our fork to use a different property name, sure.

**Comment by Gudahtt** [comment]
I just noticed that I'm seeing this problem on my fork when testing this: https://github.com/Roblox/cla-signature-bot/issues/10

I'm going to try to fix this as well. This would really clutter up the actions tab.

File: .github/workflows/cla.yml
Comment (Gudahtt) [reply .github/workflows/cla.yml]: This has been resolved here: https://github.com/MetaMask/cla-signature-bot/commit/6adffa525467ef481332325d9cc1dbbb126a3436
And the new `allowlist` parameter is in-use as of 485431ce2fa6d65b43fbb1894f988ee4cc2d63f5

**Comment by Gudahtt** [comment]
This should be ready to review again now. I've addressed the issue comment problem in 5c16eda828441fb065ec2d2d0e586ac2f63d12af by skipping the job if it was triggered by an issue comment rather than a PR comment. Unfortunately this still ends up showing up in the 'Actions' tab, but at least it's greyed out. This was the best solution I could find. It doesn't seem possible yet to trigger actions just on pull request comments but not on issue comments.

**Comment by Gudahtt** [comment]
Also I'd appreciate a review of the commits I've made on `cla-signature-bot`: https://github.com/MetaMask/cla-signature-bot/commits/master

Most of them have accompanying PRs here: https://github.com/Roblox/cla-signature-bot/pulls
The one exception is https://github.com/MetaMask/cla-signature-bot/commit/83d45d2898ee419e899b9234cbe9928224d39e0f - which isn't yet sufficiently generalized to submit back upstream.

File: .github/workflows/cla.yml
Comment (rekmarks) [inline .github/workflows/cla.yml:10]: Can you explain what the second boolean expression is checking?

Comment (Gudahtt) [reply .github/workflows/cla.yml:10]: It's checking to ensure that the URL for the comment includes the path segment `/pull/`, which it does for _pull reuest_ comments but not issue comments.

**Review by rekmarks — DISMISSED** [review]
LGTM!

**Comment by Gudahtt** [comment]
I have added one more feature to `cla-signature-bot`: [the `allowOrganizationMembers` flag](https://github.com/MetaMask/cla-signature-bot/commit/f99d1cefed3e740eb5e57206f8ba5e62c887faf1). This ensures that none of us need to sign the CLA.

**Review by rekmarks — DISMISSED** [review]

**Comment by whymarrh** [comment]
I've enabled local actions for this repo—I had disabled actions across all the repos a while back before we'd decided to start using them so as we add workflows we'll need to enable it.



Where we're using our fork I've left it at just local actions but it might even be good to try and stick to local actions for so we don't have random third-part code running on CI.

**Comment by Gudahtt** [comment]
I'm going to disable actions again for now until we get final signoff on this CLA.

I'm not sure if we need to enable third-party actions for `issue_comment` to work correctly 🤔 I suppose even PRs from forks are considered a part of this repository? I guess we'll find out.

**Comment by whymarrh** [comment]
I think that simply restricts where we can `uses` code from. That is, our `uses: MetaMask/cla-signature-bot@v3.0.0` versus `uses: roblox/cla-assistant@2.0.0`.

**Comment by Gudahtt** [comment]
Note that I've just pushed one more commit to this branch (hopefully the final one 🤞 ). I had discovered a couple of mistakes when testing, and they are now fixed.

Also I've discovered that the `allow-organization-members` flag only works for anyone who is a _public_ member of the organization. This means that any _private_ members will need to either become public or sign the CLA. This seems fine though - if anything it's a good thing, as now we aren't leaking private information via this bot's behavior.

**Review by rekmarks — APPROVED** [review]

**Comment by Gudahtt** [comment]
I have just enabled local Actions for this repository ([as described in this comment](https://github.com/MetaMask/metamask-extension/pull/9310#issuecomment-680980093)), so the bot should start posting now.

**Pull request merged**
